### ISSUE TITLE
JSON structure: array of endpoints/aliases

### ISSUE THREAD

**Issue by sftcd** [sftcd/wkesni#9]
Why don't we make the JSON value returned to $ZF from $ORIGIN to be an array where each element is either endpoint details or a alias? IOW each element of the array can have endpoint details or alias details.

The above is NOT what we say in -02, but in thinking about it, it doesn't seem to be right that a $ORIGIN has to be either all endpoints or all aliases.

Whatcha think?

Reason for the above: HTTPS or SVCB RRs can be multivalued so someone will want it.

**Comment by sftcd** [comment]
Ah - in checking I see the SVCB spec says: "SVCB RRSets SHOULD only have a single resource record in AliasMode"  - @bemasc: any insight into why that's a SHOULD and not a MUST and how we might wanna play that in this spec? (MInd you, I don't see how one can enforce that an RRset only have a single value, other than by barfing ungracefully so not sure if that SHOULD is even a good idea myself)

**Comment by bemasc** [comment]
I'm OK with allowing multiple aliases in the syntax, even though there is currently not a situation in which this is a good idea.  The text seems to allow that as well as of ccf170d5cff2af5c72d99aa37957c2c95660bba0.

**Comment by sftcd** [comment]
On 10/08/2023 20:36, Benjamin M. Schwartz wrote:
> I'm OK with allowing multiple aliases in the syntax, even though there is currently not a situation in which this is a good idea.  The text seems to allow that as well as of ccf170d5cff2af5c72d99aa37957c2c95660bba0.
> 

Ah, sorry - I made some tweaks to match the bash code and
text but wanted to test that script some more before
discussing 'em with ya in case those changes turned out bad.

I've got that script working in one basic setup now and aim
to try it out with a bit of a more complicated one tomorrow
and to then try chat about stuff like the json syntax. (What
I have right now is partly driven by what works easily with
the `jq` command which isn't a great reason for any specific
thing:-)

S.
